### DISCUSSION TITLE
eval問題

### DISCUSSION THREAD

**Discussion by Yunhans** [1101-GS3073-C/Python-Community-GS3073-C#13] [Q&A] [answered]
微積分上機的課教

    c , d , e = eval(input('分別輸入c,d,e值:'))
    print(c)
    print(d)
    print(e)

但是跑出下列

    Traceback (most recent call last):
      File "/Users/yunhans/hello-world/微積分1.py", line 25, in <module>
        c , d , e = eval(input('分別輸入c,d,e值:'))
      File "<string>", line 0
    
        ^
    SyntaxError: unexpected EOF while parsing

請問是哪裡出錯了呢

**Comment by briankuo123** [comment, marked as answer]
這個錯誤是指你可能沒有將for或while迴圈寫完整或是你有括號沒有關
我測試下來還有當你input沒有輸入任何值得時候也會跑出這個結果

https://careerkarma.com/blog/python-syntaxerror-unexpected-eof-while-parsing/

**Comment by chia-kai-chang** [comment]
`eval`是把`input`輸入的**文字**看成**數學算式**來處理

因為你的輸入是`a,b,c`，中間有逗點，它不是數學算式，所以會有錯誤訊息。

@Yunhans  嘗試把`eval`拿掉試試看~
